inductiva/resources/machine_cluster.py

```diff
@@ -0,0 +1,85 @@
+"""Class to manage the MPI cluster in Google Cloud."""
+from absl import logging
+
+from inductiva.resources import machines_base
+
+
+class MPICluster(machines_base.BaseMachineGroup):
+    """Class to launch and manage an MPI cluster in Google Cloud.
+
+    A MPI cluster is a collection of homogenous machines all working together on
+    a common task given the configurations that are launched in Google Cloud.
+    Note: The cluster will be available only after calling 'start' method.
+    The billing will start only after the machines are started."""
+
+    def __init__(self,
+                 machine_type: str,
+                 num_machines: int,
+                 disk_size_gb: int = 70,
+                 register: bool = True) -> None:
+        """Create a MPICluster object.
+
+        The register argument is used to indicate if the machine group should
+        be registered or if it was already registered. If set as False on
+        initialization, then, the machine group is not registered and it
+        can not be started in the cloud. This serves has an helper argument for
+        retrieving already registered machine groups that can be started, for
+        example, when retrieving with the `machines_groups.get` method.
+        Users should not set this argument.
+
+        Args:
+            machine_type: The type of GC machine to launch. Ex: "e2-standard-4".
+              Check https://cloud.google.com/compute/docs/machine-resource for
+              information about machine types.
+            num_machines: The number of virtual machines to launch.
```

Comment:
I'd rather have `num_vcpus` instead of `num_machines`. This option might require other classes to have the same argument renamed.

Reply:
In this case it is the actual number of machines. Notice, that a single machine can have multiple `vcpus`. Moreover, currently, we cannot assemble as we wish a group of machines to achieve a certain `num_vcpus`.